test/metrics/spanner-metrics-exporter.test.ts

```diff
@@ -0,0 +1,229 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import {MeterProvider, MetricReader} from '@opentelemetry/sdk-metrics';
+import {CloudMonitoringMetricsExporter} from '../../src/metrics/spanner-metrics-exporter';
+import {
+  GAX_METER_NAME,
+  METRIC_NAME_ATTEMPT_COUNT,
+  METRIC_NAME_ATTEMPT_LATENCIES,
+  METRIC_NAME_OPERATION_COUNT,
+  METRIC_NAME_OPERATION_LATENCIES,
+  METRIC_NAME_GFE_MISSING_HEADER_COUNT,
+  METRIC_NAME_GFE_LATENCY,
+} from '../../src/metrics/constants';
+import {Counter, Meter, Histogram} from '@opentelemetry/api';
+import {ExportResult, ExportResultCode} from '@opentelemetry/core';
+
+const PROJECT_ID = 'test_projectid';
+const INSTANCE_ID = 'test_instance';
+const DATABASE_ID = 'test_db';
+
+// Ensure custom exporter is valid
+describe('CustomExporter', () => {
+  beforeEach(() => {
+    process.env.GCLOUD_PROJECT = PROJECT_ID;
```

Comment:
Test cases are failing because we are setting `GCLOUD_PROJECT` with `test_projectid` here which is overriding the `GCLOUD_PROJECT` set to `test-project` [here](https://github.com/googleapis/nodejs-spanner/blob/main/test/spanner.ts#L314)

Reply:
I would recommend to move this to `.mocharc.js` file and also remove it from the existing test file.

Reply:
Thanks for digging into this. I'm now setting `GCLOUD_PROJECT `in `.mocharc.js` and removed instances that were defined in the tests.